homeassistant/components/unifiprotect/button.py

```diff
@@ -86,25 +121,49 @@ async def _add_new_device(device: ProtectAdoptableDeviceModel) -> None:
             data,
             ProtectButton,
             all_descs=ALL_DEVICE_BUTTONS,
+            unadopted_descs=[ADOPT_BUTTON],
             chime_descs=CHIME_BUTTONS,
             sense_descs=SENSOR_BUTTONS,
             ufp_device=device,
         )
         async_add_entities(entities)
+        _async_remove_adopt_button(hass, device)
+
+    async def _add_unadopted_device(device: ProtectAdoptableDeviceModel) -> None:
```

Comment:
This doesn't need to be a coroutine function since we don't await inside. Make it an async callback using the `@callback` decorator.